### ISSUE TITLE
oAuth sometimes loads tweets sometimes blank array

### ISSUE THREAD

**Issue by summit-web** [abraham/twitteroauth#319]
Hi there,

I'm fighting a strange issue. I've implemented OAuth fine onto a site I'm working on.

1. Sometimes it loads the tweets
2. Sometimes it returns an empty array, with one item dated 01/01/1970. (I know that's not technically an empty array, but var_dump is empty but my output gives that date, function below)

I can't figure out why it would work sometimes, but not others. no PHP Errors being dumped either inline or in error_log.

    function hlh_get_tweets(){
      $consumer_key = '';                               // Put your consumer key here
      $consumer_secret = '';         // Put your consumer secret here
      $access_token = '';   // Put your access token here
      $owner_id = ''; // just in case
      $access_token_secret = '';   // Put your access token secret here
      $nb_of_tweets = 3;                                                      // Nb of tweets to be displayed
      $include_rts = true;                                                   // true to include RT's or false to exclude them
      $connection = new TwitterOAuth($consumer_key,$consumer_secret,$access_token,$access_token_secret);
     
      /* get 'normal tweets' */                
      $tweets = $connection->get('statuses/user_timeline', array('count' => $nb_of_tweets, 'include_rts' => $include_rts, 'exclude_replies'=>true,'user_id'=>'hlhsocial','since_id'=>'572321775012454400'));
      // technically exclude_replies is not required, but good to state it for clarity
     
      /* get favourites */
      $favs = $connection->get('favorites/list',array('count'=>$nb_of_tweets,'user_id'=>'hlhsocial'));
    
      $tweetsArray = array();
      foreach($tweets as $key => $tweet){
       //array_push($tweetsArray,array(['tweet']=>$tweet->text,['id']=>$tweet->id));
       $itemId = $tweet->id_str;
       if(date("d-m-Y", strtotime($tweet->created_at))!='01/01/1970'){
        $tweetsArray[$itemId]='<div class="tweet-created-at threecol" style="text-overflow:wrap;">'.date("d-m-Y", strtotime($tweet->created_at)).'</div><div class="tweet-text ninecol">'.$tweet->text.'</div>';
       }
      }
      foreach($favs as $key => $fav){
       //array_push($tweetsArray,array(['tweet']=>$fav->text,['id']=>$fav->id));
       $itemId = $fav->id_str;
       $tweetsArray[$itemId]='<div class="tweet-created-at threecol">'.date("d-m-Y", strtotime($fav->created_at)).'</div><div class="tweet-text ninecol">'.$fav->text.'</div>';
      }
      $sortedTweets = krsort($tweetsArray);
      
      /* output */
      $tweetOutput = '';
      $i=0;
      foreach($tweetsArray as $tweet){
        if($i==3) break;
       $tweetOutput .= '<li class="tweet">'.preg_replace('/@([a-zA-Z0-9_]+)/','<a href="http://twitter.com/$1" rel="nofollow">@$1</a>',preg_replace('@(https?://([-\w.]+[-\w])+(:\d+)?(/([\w-.~:/?#\[\]\@!$&\'()*+,;=%]*)?)?)@', '<a href="$1" target="_blank" rel="nofollow">$1</a>', $tweet)) . '</li>';
       $i++;
      }
      return '<ul class="tweet-list">'.$tweetOutput.'</ul>';
     } 
    }

**Comment by abraham** [comment]
You have `nb_of_tweets` = 3 and `exclude_replies` = true. The 12 most recent tweets for hlhsocial are all replies so you would have to bump count up to at least 13 to get a single tweet back.

hlhsocial is a `screen_name` not a `user_id` so you should change the param name.

You should only be running each foreach loop if `$connection->getLastHttpCode() == 200`. Otherwise Twitter might have returned an error and you could be looping over an error response.

You should really only use `since_id` if you are using pagination.

**Comment by summit-web** [comment]
I was using since_id as a few entries when I started developing this were returning 01/01/1970 so I started the tweets from one of theirs at the start of March.

I had to set `exclude_replies` as this is a published tweet feed on a website, but the client didn't want 'conversational' tweet included - hence merging normal tweets and favourite.

Increasing the tweet limit and amending `screen_name` seems to have fixed it. :) Thanks so much for your quick reply!
